src/measure_theory/conditional_expectation.lean

```diff
@@ -262,4 +262,62 @@ end complete_subspace
 
 end Lp_meas
 
+section integral_norm_le
+
+variables {m m0 : measurable_space α} {μ : measure α} {s : set α}
+
+/-- Let `m` be a sub-σ-algebra of `m0`, `f` a `m0`-measurable function and `g` a `m`-measurable
+function, such that their integrals coincide on `m`-measurable sets with finite measure.
+Then `∫ x in s, ∥g x∥ ∂μ ≤ ∫ x in s, ∥f x∥ ∂μ` on all `m`-measurable sets with finite measure. -/
+lemma integral_norm_le_of_forall_fin_meas_integral_eq (hm : m ≤ m0) {f g : α → ℝ}
+  (hf : measurable f) (hfi : integrable_on f s μ) (hg : measurable[m] g) (hgi : integrable_on g s μ)
+  (hgf : ∀ t, measurable_set[m] t → μ t ≠ ∞ → ∫ x in t, g x ∂μ = ∫ x in t, f x ∂μ)
```

Comment:
Heather has convincingly argued that in this kind of situation the assumption on `μ t` should be spellt as `μ t < ∞`, see https://github.com/leanprover-community/mathlib/pull/8300#discussion_r669834882